src/MiniCover/Commands/Options/MiniCoverOption.cs

```diff
@@ -5,36 +5,33 @@ namespace MiniCover.Commands.Options
 {
     public abstract class MiniCoverOption<TValue> : IMiniCoverOption<TValue>
     {
-        private CommandOption option;
-        private bool validated;
-        private TValue value;
+        protected CommandOption Option;
+        protected bool Validated { get; private set; }
+        protected TValue ValueField { get; private set; }
 
-        protected abstract string Description { get; }
-        protected abstract string OptionTemplate { get; }
-        protected virtual CommandOptionType Type => CommandOptionType.SingleValue;
-
-        public TValue Value
+        public TValue GetValue()
         {
-            get
-            {
-                if (!validated)
-                    throw new MemberAccessException("Option should be validated before Value access");
+            if (!Validated)
+                throw new MemberAccessException("Option should be validated before GetValue access");
 
-                return value;
-            }
+            return ValueField;
         }
 
+        protected abstract string Description { get; }
+        protected abstract string OptionTemplate { get; }
+        protected virtual CommandOptionType Type => CommandOptionType.SingleValue;
+
         public virtual void AddTo(CommandLineApplication command)
         {
-            option = command.Option(OptionTemplate, Description, Type);
+            Option = command.Option(OptionTemplate, Description, Type);
         }
 
-        public void Validate()
+        public virtual void Validate()
```

Comment:
instead of overriding something needed:

Better to have :
```csharp
        public void Validate()
        {
            ValueField = GetOptionValue();
            Validated = Validation();
        }

        protected virtual bool Validation() => true;
```